### PR TITLE
fix(pose): 女の子座りの生成品質・APIエラーのUI表示を改善

### PR DESCRIPTION
**Pull request by koizuka**
## 概要

ポーズ生成の品質改善第2弾。「女の子座り」と入力しても股関節が内旋したポーズにならない問題を、実機での反復検証(JSON確認・真横視点での切り分け)を踏まえて根本から修正しました。あわせて、API エラーの内容がコンソールを開かないと分からなかった問題を解消します。

### 女の子座り(ぺたん座り / W-sitting)対応
- **股関節 `rotation` クランプの対称化(-30..90 → -90..90)** — 女の子座りに必要な内旋 -45〜-60° は、モデルが正しく出力してもサニタイザで -30 に切られていた
- **`LegPose.shinTwist`(脛骨回旋)を新設** — 膝下ボーンの Y 回転を屈曲(X)より先に適用することで、シンの軸方向を変えずに足先の向きだけを回す純粋な捻りとして実装。深い股関節内旋で爪先が内に巻き込む問題への対処。正座・横座り等の足先調整にも流用可能
- **`kneeBend` 上限 150→160°**(踵が尻に付く実際の可動域)
- **プロンプトに女の子座りレシピを追加** — 深い内旋 -65〜-80(膝の折り面を床に寝かせてシンが床に沿って外へ流れる)+ shinTwist 25-40 + 上体完全直立(体と脚での股関節角の辻褄合わせを明示的に禁止)+ 膝置きの腕

### 腕の自然なデフォルト
- 腕未指定時にVRMレスト(Tポーズ)のまま放置されていたのを、`RELAXED_ARM`(体からわずかに離した下げ腕)の適用に変更
- プロンプトの NATURAL POSE BIAS に「何もしていない腕は省略せよ(省略=自然な下げ腕として描画)」を追記

### API エラーの UI 表示 / thinking 再有効化
- `thinking: {type:'disabled'}` を撤回し adaptive thinking を有効化(空間推論に有効)。当時の問題(思考が max_tokens を食い尽くす)は `max_tokens` 2048→8192 で対処
- `stop_reason: 'max_tokens'` を `AnthropicTruncatedError` として検出し、パースエラーに化けていた失敗を原因どおりに表示
- API エラー JSON の `error.message` を各エラークラスの `detail` に保持し、エラー Alert の下に等幅小フォントで表示

## テスト

- `npm run lint` ✅
- `npm run build` ✅
- `npm run test` ✅ 665件パス(shinTwist の軸不変性/外旋方向、クランプ、truncated エラー、エラー detail 抽出などを追加)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

> [!NOTE]
> **Medium Risk**
> Touches core pose kinematics (leg Euler order, defaults) and live Anthropic request/parse behavior; regressions could affect many seated poses or generation reliability, though coverage is expanded in tests.
> 
> **Overview**
> Improves **girl-style / W-sitting** poses and makes Claude failures easier to diagnose in the pose panel.
> 
> **Pose model & prompts:** Adds **`shinTwist`** (tibial rotation on the lower leg with explicit Euler order) so feet can aim without changing shin direction. Hip **`rotation`** sanitization is symmetric **−90..90** (deep internal rotation was previously clamped), **`kneeBend`** max is **160°**, and prompts/schema document **onnanoko-zuwari** (deep internal rotation, shinTwist, upright torso). Omitted arms now map to **`RELAXED_ARM`** instead of T-pose rest; the prompt tells the model to omit idle arms.
> 
> **Anthropic client & UI:** Re-enables adaptive thinking and raises **`max_tokens` to 8192**; **`stop_reason: max_tokens`** becomes **`AnthropicTruncatedError`** with i18n **`anthropicTruncated`** (still accepts replies where JSON finished before the cutoff). API errors carry **`detail`** from `error.message`; **`PoseSourcePanel`** shows it under the alert in monospace.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit faa8d6aa64153578bfc871e0599d6751d7df53d9. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

---
## Summary by cubic
Improves W‑sitting generation with deep hip internal rotation and tibial twist, makes omitted arms render relaxed, and surfaces Anthropic API errors with detail. Re‑enables adaptive thinking with a larger token budget and robust truncation handling.

- **New Features**
  - Added `shinTwist` (tibial rotation) with explicit 'XYZ' order so feet turn without moving the shin; fixes inward‑curling toes in W‑sitting.
  - Unspecified arms now render as relaxed hanging (`RELAXED_ARM`); prompt/schema updated with onnanoko‑zuwari recipe and `shinTwist`.
  - Adaptive thinking re‑enabled with `max_tokens` 8192.

- **Bug Fixes**
  - Hip rotation clamp made symmetric (−90..90) and `kneeBend` max raised to 160 for deep internal rotation.
  - `stop_reason: max_tokens` is handled correctly: accept replies with complete pose JSON; otherwise throw `AnthropicTruncatedError` instead of a parse error.
  - API errors now show a clear alert with server `error.message` detail; added `anthropicTruncated` i18n.

<sup>Written for commit faa8d6aa64153578bfc871e0599d6751d7df53d9. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/koizuka/drawing-practice/pull/260?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: src/utils/anthropic.ts [hunk 1]
```diff
@@ -158,16 +200,26 @@ export async function generatePose(pngBase64: string | null, hint: string, signa
...
   // Debug-level so pose tuning can inspect the model's analysis + JSON from
   // the console (Safari Web Inspector) without noisy default output.
   console.debug('[pose] model reply:', text);
+  // Cut off before (or mid-) JSON: without this check the failure surfaces
+  // as an unhelpful parse error. stop_reason is checked before parsing so
+  // the user sees the actual cause (budget exhausted, incl. thinking).
+  if (data.stop_reason === 'max_tokens') {
+    console.error('[pose] reply truncated by max_tokens:', data);
+    throw new AnthropicTruncatedError('stop_reason: max_tokens');
+  }
```
[the remark this answers is not in the document]
Comment (koizuka) [reply src/utils/anthropic.ts]: Fixed in faa8d6a — parsing now happens first; `AnthropicTruncatedError` is thrown only when no usable pose JSON can be extracted (plus the edge case where a truncated reply "parses" into an empty pose via a stray complete object in the prose). Added tests for both the complete-JSON-then-max_tokens and truncated-with-stray-object cases.

File: src/pose/poseMapping.ts [hunk 2]
```diff
@@ -147,7 +155,14 @@ function applyLeg(resolve: BoneResolver, sideName: Side, leg: LegPose): void {
...
   );
-  resolve(`${sideName}LowerLeg`)?.rotation.set((leg.kneeBend ?? 0) * DEG, 0, 0);
+  // Default 'XYZ' order applies Y before X, so shinTwist is a pure twist
+  // about the shin's own bone axis (tibial rotation carried through the
+  // bend) — it reorients the foot without moving the shin's direction.
+  resolve(`${sideName}LowerLeg`)?.rotation.set(
+    (leg.kneeBend ?? 0) * DEG,
+    side * (leg.shinTwist ?? 0) * DEG,
+    0,
+  );
```
[the remark this answers is not in the document]
Comment (koizuka) [reply src/pose/poseMapping.ts]: Fixed in faa8d6a — the Euler order 'XYZ' is now explicit on the lower-leg rotation.set, with the comment updated to note the kinematics depend on it.

[the remark this answers is not in the document]
Comment (koizuka) [reply src/pose/poseMapping.ts:167]: Fixed in faa8d6a — made the 'XYZ' order explicit as suggested.
